src/newton_ac_powerflow.jl

```diff
@@ -0,0 +1,731 @@
+    ts_V = zeros(Complex{Float64}, length(data.bus_type[:, 1]), length(sorted_time_steps))
+    ts_S = zeros(Complex{Float64}, length(data.bus_type[:, 1]), length(sorted_time_steps))
+
+    # TODO If anything in the grid topology changes, 
+    #  e.g. tap positions of transformers or in service 
+    #  status of branches, Yft and Ytf must be updated!
+    Yft = data.power_network_matrix.yft
+    Ytf = data.power_network_matrix.ytf
+    fb = data.power_network_matrix.fb
+    tb = data.power_network_matrix.tb
+
+    for t in sorted_time_steps
+        converged, V, Sbus_result =
+            _ac_powereflow(data, pf; time_step = t, kwargs...)
+        ts_converged[t] = converged
+        ts_V[:, t] .= V
+        ts_S[:, t] .= Sbus_result
+
+        if converged
+            ref = findall(
+                x -> x == PowerSystems.ACBusTypesModule.ACBusTypes.REF,
+                data.bus_type[:, t],
+            )
+            pv = findall(
+                x -> x == PowerSystems.ACBusTypesModule.ACBusTypes.PV,
+                data.bus_type[:, t],
+            )
+            pq = findall(
+                x -> x == PowerSystems.ACBusTypesModule.ACBusTypes.PQ,
+                data.bus_type[:, t],
+            )
+
+            # temporary implementation that will need to be improved:
+            # write results for REF
+            data.bus_activepower_injection[ref, t] .=
+                real.(Sbus_result[ref]) .+ data.bus_activepower_withdrawals[ref, t]
+            data.bus_reactivepower_injection[ref, t] .=
+                imag.(Sbus_result[ref]) .+ data.bus_reactivepower_withdrawals[ref, t]
+            # write Q results for PV
+            data.bus_reactivepower_injection[pv, t] .=
+                imag.(Sbus_result[pv]) .+ data.bus_reactivepower_withdrawals[pv, t]
+            # results for PQ buses do not need to be updated -> already consistent with inputs
+
+            # write voltage results
+            data.bus_magnitude[pq, t] .= abs.(V[pq])
+            data.bus_angles[pq, t] .= angle.(V[pq])
+            data.bus_angles[pv, t] .= angle.(V[pv])
+        else
+            data.bus_activepower_injection[:, t] .= NaN64
+            data.bus_activepower_withdrawals[:, t] .= NaN64
+            data.bus_reactivepower_injection[:, t] .= NaN64
+            data.bus_reactivepower_withdrawals[:, t] .= NaN64
+            data.bus_magnitude[:, t] .= NaN64
+            data.bus_angles[:, t] .= NaN64
+        end
+    end
+
+    # write branch flows
+    # TODO if Yft, Ytf change between time steps, this must be moved inside the loop!
+    Sft = ts_V[fb, :] .* conj.(Yft * ts_V)
+    Stf = ts_V[tb, :] .* conj.(Ytf * ts_V)
+
+    data.branch_activepower_flow_from_to .= real.(Sft)
+    data.branch_reactivepower_flow_from_to .= imag.(Sft)
+    data.branch_activepower_flow_to_from .= real.(Stf)
+    data.branch_reactivepower_flow_to_from .= imag.(Stf)
+
+    data.converged .= ts_converged
+
+    return
+end
+
+function _ac_powereflow(
```

Comment:
@rbolgaryn fix typo:
```suggestion
function _ac_powerflow(
```